src/commands/project/__tests__/add.test.js

```diff
@@ -1,4 +1,22 @@
 // @flow
-import { projectAdd, toProjectAddOptions } from '../add';
+import { projectAdd } from '../add';
+import { toAddOptions } from '../../add';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as yarn from '../../../utils/yarn';
+import fixtures from 'fixturez';
 
-test('bolt project add');
+const f = fixtures(__dirname);
+
+jest.mock('../../../utils/logger');
+jest.mock('../../../utils/yarn');
+
+describe('bolt project add', () => {
+  test('adding a project dependency only used by the project', async () => {
+    let tempDir = f.copy('package-with-external-deps-installed');
+
+    await projectAdd(toAddOptions(['project-new-dep'], { cwd: tempDir }));
+
+    expect(yarn.add).toHaveBeenCalledTimes(1);
```

Comment:
Sorry, don't want to nitpick, but can this check that it was called with the correct cwd too?

Reply:
actually, this is a good catch. There is a problem in completely reusing the add.

Reply:
@sendilkumarn Can you please elaborate what is problem coming with this approach?

Reply:
`yarn add` should happen in the pkg directory rather than the `project` or `project.pkg`

We should not set the dependencies. Also, there is no symlink needed here. (since both will be handled by the yarn)

Reply:
I think it should be add it to `project`, what you did is correct. 

for symlink **(since both will be handled by the yarn)**,  I am not sure what so you mean by that?

@lukebatchelor what do you think?

@sendilkumarn 
If you are interested we can continue this discussion on slack - https://join.slack.com/t/boltpkg/shared_invite/enQtMjkzNDk4NjI0OTYyLThhYWRmMDRiMGNiNzEzNWYwYjMwMzJiYjk3ZWE4NGM4N2ZlODA4YzY4NWU5MGM3MDUxNDU5Mzk5ZDg4NTRlODA ?

Reply:
awesome let me join

Reply:
Sorry, might have been confusion here.

We *do* need to use `addDependenciesToPackage` as you have below. 

All `bolt project [cmd]`'s should just be the equivalent to running `cd path/to/project/dir && bolt [cmd]`. The difference here would be the extra checks that `addDependenciesToPackage` does (checking internal/external deps, the side effect of bumping any workspace deps if its changed a version of an external dep, etc).

The test should probably check that `addDependenciesToPackage` is called then, rather than `yarn.add`

Reply:
I completely agree with this

But,

`await yarn.add(project.pkg, externalDepsToInstallForProject, type);` 
The above considers `package` (not project).This method is in fact written on the assumption that wherever we are in the project, do the instructions on the root of the package. 

But `bolt project [cmd]` says don't worry about the `package` do things from the directory where the command is executed.

Reply:
> But bolt project [cmd] says don't worry about the package do things from the directory where the command is executed.

Bolt project cmd, should run in the project dir not in the dir it's executed. See my comment above.

Reply:
Sorry I missed the project word there :)